scipy/_lib/tests/test__gcutils.py

```diff
@@ -75,7 +75,7 @@ class C:
         # to function correctly.  It is assigned to c, but c itself is
         # not referenced in the body of the with, it is only there for
         # the refcount.
-        with assert_deallocated(C) as c:
+        with assert_deallocated(C):
```

Comment:
This change causes the failure
```
================================================================== FAILURES ==================================================================
_______________________________________________________ test_assert_deallocated_nodel ________________________________________________________
[gw3] linux -- Python 3.10.8 /home/warren/py3.10.8/bin/python3
scipy/_lib/tests/test__gcutils.py:72: in test_assert_deallocated_nodel
    with pytest.raises(ReferenceError):
E   Failed: DID NOT RAISE <class 'scipy._lib._gcutils.ReferenceError'>
        C          = <class 'scipy._lib.tests.test__gcutils.test_assert_deallocated_nodel.<locals>.C'>
========================================================== short test summary info ===========================================================
FAILED scipy/_lib/tests/test__gcutils.py::test_assert_deallocated_nodel - Failed: DID NOT RAISE <class 'scipy._lib._gcutils.ReferenceError'>
============================================= 1 failed, 86 passed, 2 skipped, 1 xfailed in 8.03s =============================================

```
It looks like the test requires the reference to be held by `c`.

A possible fix that makes `ruff` happy is
```
        with assert_deallocated(C) as _:
```

Reply:
Thanks!